### ISSUE TITLE
[Enhancement] Replace Python 3.4.3 with 3.6.0

### ISSUE THREAD

**Issue by suic86** [Codewars/codewars-runner-cli#296]
Python 3.5 and 3.6 contains a bunch of interesting new features (format string, enhancement to asynchronous programming). It would be nice to have these available on CW.

Regards,

suic

**Comment by CaptainSpalding** [comment]
For sure, I have wrote a lot of my codes inn this fashion before the enhancement, for years actually. I have always refered  to it as "Parallel" they call it "ASYNC". It is awesome, and I don't have to write a lot of extra code to implement this functionality anymore as it is lib standard in python now :)

**Comment by kazk** [comment]
@suic86 

> Python 3.6.0 is the newest major release of the Python language, and it contains many new features and optimizations.

Is there any backward incompatible changes that you know? If so, we need a way to keep Python 3.4.3 along with it.

---

- Python 3.5
  - [What’s New In Python 3.5](https://docs.python.org/3.5/whatsnew/3.5.html)
  - [Python3.5.3 (2017-01-17)](https://www.python.org/downloads/release/python-353/)
- Python 3.6
  - [What’s New In Python 3.6](https://docs.python.org/3.6/whatsnew/3.6.html)
  - [Python3.6.0 (2016-12-23)](https://www.python.org/downloads/release/python-360/)
- Installing
  - [Installing Python3.6 using apt-get on Ubuntu 14.04](http://askubuntu.com/a/865569)
  - [pyenv](https://github.com/yyuu/pyenv)

**Comment by suic86** [comment]
@kazk I don't know about any major backward incompatible change and the descriptions in _What's New_ documents don't contain any major incompatibilities. I think that it's safe to upgrade to Python 3.6. 

Regards,

suic

**Comment by kazk** [comment]
Thanks. I don't know when I can try this because `codewars/python-runner` was over 800MB when I built last time and building it made my old machine unusable for a while. Maybe someone with better machine/network can try it.

---

I've been thinking how to do the "Paradigm Shift" (under TODO in README) for the past few weeks.
I still need to ask @jhoffner about the requirements for the use in `runner.codewars.com` and if he's interested or not, but I have some ideas to easily switch environments, reduce the image sizes and lower the barrier of contribution.
If this idea works, having Python 2.7/3.4/3.5/3.6 at the same time is easy and any future versions can be added without worrying about backward compatibility.

**Comment by kazk** [comment]
I just realized that simply installing Python 3.6 won't benefit Codewars users because I don't think the [test framework](https://github.com/Codewars/codewars-runner-cli/blob/master/frameworks/python3/cw-2.py) used by Codewars supports asynchronous tests 😞

**Comment by kazk** [comment]
@suic86 Python 3.6 is now available.
I misunderstood Python's asynchronous programming which lead to my comment above (I still don't know much about it). Sorry for the confusion.

The following code runs in kumite, but I don't know what I'm doing.

```python
import asyncio

async def async_add(x, y):
  await asyncio.sleep(1)
  return x + y
```

```python
import asyncio

async def test_async(co, v):
  test.assert_equals(await co, v)
  
loop = asyncio.get_event_loop()
loop.run_until_complete(asyncio.gather(
  test_async(async_add(1, 1), 2),
  test_async(async_add(1, 2), 3),
  test_async(async_add(1, 3), 4)
))
loop.close()
```

**Comment by CaptainSpalding** [comment]
setting a time gap then adding x + y every second, you need to await an add
function and spit it out from another function. it is not threading
depending on size multiple actions can be carried out on a single thread,
as far as async add method goes, propbably no noticible speed diff unless
numbers were psychotically huge

**Comment by kazk** [comment]
@CaptainSpalding I don't think I understand what you mean. Can you post an example code? Like I wrote above, I don't have any experience using this. I just read the docs quickly and simplified one of the examples to show that it runs on Codewars environment.

Here's what I *think* what I'm doing:

- `await asyncio.sleep(1)` simulates an operation taking some time, so each `async_add` takes at least a second in this case.
- `test_async` calls `test.assert_equals` with the result and expected value once `async_add` finishes. `test.assert_equals(await async_add(1, 1), 2)`
- `loop.run_until_complete(asyncio.gather())` executes tasks in parallel.

Please let me know if I'm misunderstanding anything. Thanks for your comment.
